docs/design/satellite-svc-separation.md

```diff
@@ -0,0 +1,198 @@
+# Satellite Service Separation
+
+## Abstract
+
+The goal of the design doc is to describe the work necessary to make the Satellite horizontally scalable.
+
+## Background
+
+Currently all Satellite services run in a single binary in a single container. While this is great for development, this is bad for scaling.
+
+### Current Satellite design
+
+Currently the Satellite is a single binary made of up the following services:
+
+#### overlay
+wip
+
+#### metainfo
+Metainfo is responsible for all things related to the metainfo stored for each file on the network. The metainfo system is currently composed of the following parts: 1) metainfoDB, a key/value store with segment path names as the key and a pointer/segment object as the value, 2) metainfo loop which iterates over all the key/values in metainfDB, 3) metainfo endpoint which creates the public GRPCs for creating/deleting metainfo, 4) metainfo service.
```

Comment:
Here's the definitions I've been using:

* `metainfo` - the concept of meta-information about the buckets, objects, streams and segments
* `satellite/metainfo.DB` - the database storing meta-information
* `satellite/metainfo.Loop` - service for iterating over the meta-information
* `pointer db` - key-value store that has no concept of buckets and objects for storing pointers

Reply:
Yeah, that makes sense. The reason I didn't define `metainfo` in terms of pointer db is because that is more of an implementation detail and may change and it uses other tables / databases besides pointer db.

Reply:
Also from end-user perspective "storing of metainfo" is more important concept, so it is probably more stable with regards to definition in the long run.